core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -52,9 +53,13 @@ object PythonRunner {
     // Format python file paths before adding them to the PYTHONPATH
     val formattedPythonFile = formatPath(pythonFile)
     val formattedPyFiles = resolvePyFiles(formatPaths(pyFiles))
+    lazy val isAPIModeClassic = sparkConf.getOption(
+      "spark.api.mode").exists(_.toLowerCase(Locale.ROOT) == "classic")
+    lazy val isAPIModeConnect = sparkConf.getOption(
+      "spark.api.mode").exists(_.toLowerCase(Locale.ROOT) == "connect")
 
     var gatewayServer: Option[Py4JServer] = None
-    if (sparkConf.getOption("spark.remote").isEmpty) {
+    if (sparkConf.getOption("spark.remote").isEmpty || isAPIModeClassic) {
```

Comment:
Can you please throw exceptions when you encounter incompatible configurations. For example: Classic should only work with master set, and not with remote set.

Reply:
That is actually thrown (fail fast) at `SparkSubmitArguments.validateSubmitArguments`